File: DDD DSL/DDD DSL Design War FizzBuzzWhizz Reloaded/Code/Java/src/fayelab/ddd/fbwreloaded/complete/interpreter/Action.java
package fayelab.ddd.fbwreloaded.complete.interpreter;

public class Action
{
    enum Type {TOFIZZ, TOBUZZ, TOWHIZZ, TOSTR};

    private Type type;

    public Action(Type type)
    {
        this.type = type;
    }

    public Type getType()
    {
        return type;
    }
}

File: DDD DSL/DDD DSL Design War FizzBuzzWhizz Reloaded/Code/Python/oo/action.py
class Action(object):
    def act(self, n):
        pass

class ToFizz(Action):
    def act(self, n):
        return 'Fizz'

class ToBuzz(Action):
    def act(self, n):
        return 'Buzz'

class ToWhizz(Action):
    def act(self, n):
        return 'Whizz'

class ToStr(Action):
    def act(self, n):
        return str(n)